Author: Agustin Martin Domingo <agmartin@debian.org>
Description: Protect against missing (multibyte-string-p) and (string-to-multibyte str) for XEmacs


--- a/support/emacsen/ispell.el
+++ b/support/emacsen/ispell.el
@@ -1648,11 +1648,14 @@ Protects against bogus binding of `enabl
 		(error "No data for dictionary \"%s\", neither in `ispell-local-dictionary-alist' nor in `ispell-dictionary-alist'"
 		       ispell-current-dictionary)))
 	 (str (nth n slot)))
-    (when (and (> (length str) 0)
-	       (not (multibyte-string-p str)))
-      (setq str (ispell-decode-string str))
-      (or (multibyte-string-p str)
-	  (setq str (string-to-multibyte str))))
+    ;; XEmacs: Not available `multibyte-string-p' `string-to-multibyte'
+    (if (featurep 'xemacs)
+	(setq str (ispell-decode-string str))
+      (when (and (> (length str) 0)
+		 (not (multibyte-string-p str)))
+	(setq str (ispell-decode-string str))
+	(or (multibyte-string-p str)
+	    (setq str (string-to-multibyte str)))))
     str))
 
 (defun ispell-get-casechars ()
